libtransmission/rpc-server.cc

```diff
@@ -643,6 +651,41 @@ void rpc_server_start_retry_cancel(tr_rpc_server* server)
     server->start_retry_counter = 0;
 }
 
+#ifdef WITH_OPENSSL
+struct bufferevent* bevcb(struct event_base* base, void* arg)
+{
+    struct bufferevent* r = nullptr;
+    auto ctx = reinterpret_cast<SSL_CTX*>(arg);
+    r = bufferevent_openssl_socket_new(base, -1, SSL_new(ctx), BUFFEREVENT_SSL_ACCEPTING, BEV_OPT_CLOSE_ON_FREE);
+    return r;
+}
+
+SSL_CTX* tr_set_cert(char const* cert, char const* key)
```

Comment:
This function doesn't set anything but returns the newly allocated context meant to be set somewhere else, so its current name is misleading. It also seems that you're leaking the context in case of errors here.

Reply:
Actually SSL context is set in this function so I don't think func name is misleading. I added `SSL_CTX_free(ctx); ` to avoid memory leak.

Reply:
I disagree. The usual understanding for `*set*` functions is to operate on a user-provided object, be it e.g. implicit `this` (for methods within a class) or an object passed in explicitly as an argument. None of it happens here, the function instead creates a new object and returns is, so whatever configuration it performs on this new object internally is its own business which as far as function users are concern is just initial setup on that new object while the main purpose of the function is to produce a new object, not modify an existing one. Note the difference between "setting" and "creating and setting up" something.

Also note that functions with `tr_` prefix are meant to be public interface functions provided to the user, which is not the case here.

Reply:
Sorry my bad. I adjusted func name again.